### ISSUE TITLE
selenium-webdriver RubyGem: can't modify frozen String

### ISSUE THREAD

**Issue by rlue** [SeleniumHQ/selenium#7365]
## 🐛 Bug Report

In the `selenium-webdriver` RubyGem (branch `rb-3.x`), certain files [use the "shovel operator" (`<<`) to modify strings in-place](https://github.com/SeleniumHQ/selenium/blob/rb-3.x/rb/lib/selenium/webdriver/firefox/binary.rb#L134), while also [containing a `frozen_string_literal: true` comment at the top of the file](https://github.com/SeleniumHQ/selenium/blob/rb-3.x/rb/lib/selenium/webdriver/firefox/binary.rb#L1) in anticipation of Ruby 3.x compatibility. (Or at least, I've found it in one file and am supposing it might be present in other files.)

Having both of these in the same file leads to a `FrozenError: can't modify frozen String` exception when the given code path is reached.

This appears to be [fixed on the `master` branch](https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/firefox/binary.rb#L56-L57), but not in the latest release of the RubyGem (3.142.3).

## To Reproduce

```ruby
>> require 'selenium-webdriver'
>> FileUtils.touch('foo')
>> FileUtils.chmod('+x', 'foo')
>> Selenium::WebDriver::Firefox::Binary.path=('foo')
>> FileUtils.rm_rf('foo')
>> Selenium::WebDriver::Firefox::Binary.path
FrozenError: can't modify frozen String
```

## Expected behavior

`Selenium::WebDriver::Firefox::Binary.path` should raise the appropriate error message instead:

```
Could not find Firefox binary (os=#{Platform.os}). Make sure Firefox is installed or set the path manually with #{self}.path=
```

## Test script or set of commands reproducing this issue

See above.

## Environment

OS: Debian (testing)
Browser: N/A
Browser version: N/A
Browser Driver version: N/A
Language Bindings version: 3.142.3
Selenium Grid version (if applicable): N/A

## Note

FWIW my preferred style for long strings is to use a heredoc:

```ruby
raise Error::WebDriverError, <<~MSG
  Could not find Firefox binary (os=#{Platform.os}).
  Make sure Firefox is installed or set the path manually with #{self}.path=
MSG
```

This is different from the fix on `master` in that it includes a `\n` between the two sentences of the error message, but I personally find it more readable.

**Comment by snowblink** [comment]
What do we need to get a gem released?

**Comment by p0deje** [comment]
This has been fixed and released in 3.142.4. Thank you for the report!
